File: agent/bindings/csharp/ModelRef.cs
using System.Text.Json.Serialization;

namespace Cody.Core.Agent.Protocol
{
  public class ModelRef
  {
    [JsonProperty(PropertyName = "providerId")]
    public ProviderId ProviderId { get; set; }
    [JsonProperty(PropertyName = "apiVersionId")]
    public ApiVersionId ApiVersionId { get; set; }
    [JsonProperty(PropertyName = "modelId")]
    public ModelId ModelId { get; set; }
  }
}

File: agent/bindings/kotlin/lib/src/main/kotlin/com/sourcegraph/cody/agent/protocol_generated/ModelRef.kt
@file:Suppress("FunctionName", "ClassName", "unused", "EnumEntryName", "UnusedImport")
package com.sourcegraph.cody.agent.protocol_generated;

import com.google.gson.annotations.SerializedName;

data class ModelRef(
  val providerId: String,
  val apiVersionId: ApiVersionIdEnum, // Oneof: unknown
  val modelId: String,
) {

  enum class ApiVersionIdEnum {
    @SerializedName("unknown") Unknown,
  }
}

